RUFAS/biophysical/manure/storage/composting.py

```diff
@@ -0,0 +1,574 @@
+from copy import copy
+from enum import Enum
+import math
+from RUFAS.biophysical.manure.storage.storage import Storage
+from RUFAS.biophysical.manure.storage.storage_cover import StorageCover
+from RUFAS.current_day_conditions import CurrentDayConditions
+from RUFAS.data_structures.animal_to_manure_connection import ManureStream
+from RUFAS.time import Time
+from RUFAS.units import MeasurementUnits
+
+
+class CompostingType(Enum):
+    """
+    This is an Enum class that represents different types of composting.
+
+    Attribute
+    ----------
+    INTENSIVE_WINDROW : str
+        Intensive windrow.
+    PASSIVE_WINDROW : str
+        Passive windrow.
+    STATIC_PILE: str
+        Static pile.
+
+    """
+
+    INTENSIVE_WINDROW = "intensive windrow"
+    PASSIVE_WINDROW = "passive windrow"
+    STATIC_PILE = "static pile"
+
+
+ACHIEVABLE_METHANE_EMISSION: float = 0.24
+"""Achievable emission of methane from dairy manure (m^3 methane / kg volatile solids)."""
+
+MCF_COMPOSTING_STATIC_PILE: float = 0.005
+"""The MCF for static pile composting."""
+
+MCF_LOWER_BOUND_TEMPERATURE: float = 15.0
+"""The lower bound temperature for determining MCF for windrow composting."""
+
+MCF_UPPER_BOUND_TEMPERATURE: float = 25.0
+"""The lower bound temperature for determining MCF for windrow composting."""
+
+MCF_COMPOSTING_WINDROW_LOW: float = 0.005
+"""The MCF for windrow composting when the air temperature is below the lower bound temperature."""
+
+MCF_COMPOSTING_WINDROW_MEDIUM: float = 0.01
+"""The MCF for windrow composting when the air temperature is between the lower and upper bound temperature."""
+
+MCF_COMPOSTING_WINDROW_HIGH: float = 0.015
+"""The MCF for windrow composting when the air temperature is above the upper bound temperature."""
+
+DEFAULT_CARBON_FRACTION_AVAILABLE_IN_MANURE: float = 0.5
+"""Default proportion of carbon available in manure, (unitless, [0, 1]). Default is set to 0.5."""
+
+DEFAULT_CARBON_FRACTION_AVAILABLE_IN_BEDDING: float = 0.35
+"""Default proportion of carbon available in bedding, (unitless, [0, 1]). Default is set to 0.35."""
+
+DEFAULT_EFFECT_OF_MOISTURE_ON_MICROBIAL_DECOMPOSITION: float = 0.65
+"""The default effect of moisture on microbial decomposition."""
+
+DEFAULT_FIRST_ORDER_DECAYING_COEFFICIENT: float = 0.1
+"""The default first order decaying coefficient."""
+
+DEFAULT_LAG_TIME: int = 2
+"""Default lag time used in the calculation of the carbon decomposition rate (days). Default is set to 2."""
+
+DEFAULT_DAYS_SINCE_LAST_TILLAGE: int = 1
+"""Default days since last tillage (days). Default is set to 1."""
```

Comment:
```suggestion
DEFAULT_DAYS_SINCE_LAST_TURNING: int = 1
"""Default days since the previous compost turning event (days). Default is set to 1."""
```
As discussed in the weekly meeting today, even though the value of this parameter may be the same in open lot, CBPB, and composting, I think we should make a specific constant for each, so that we can eventually implement user input here instead of a constant/default.